inst/include/model_simulation.hpp

```diff
@@ -0,0 +1,135 @@
+          pars.incidence_relative_risk_age(a - pars.adult_incidence_first_age_group, g, time_step) *
+          intermediate.hiv_negative_pop(a, g);
+    }
+  }
+
+  intermediate.incidence_rate_sex(MALE) =
+      pars.incidence_rate(time_step) * (intermediate.hiv_neg_aggregate(MALE) + intermediate.hiv_neg_aggregate(FEMALE)) /
+      (intermediate.hiv_neg_aggregate(MALE) +
+       pars.incidence_relative_risk_sex(time_step) * intermediate.hiv_neg_aggregate(FEMALE));
+  intermediate.incidence_rate_sex(FEMALE) =
+      pars.incidence_rate(time_step) * pars.incidence_relative_risk_sex(time_step) *
+      (intermediate.hiv_neg_aggregate(MALE) + intermediate.hiv_neg_aggregate(FEMALE)) /
+      (intermediate.hiv_neg_aggregate(MALE) +
+       pars.incidence_relative_risk_sex(time_step) * intermediate.hiv_neg_aggregate(FEMALE));
+
+  for (int g = 0; g < pars.num_genders; g++) {
+    for (int a = pars.hiv_adult_first_age_group; a < pars.age_groups_pop; a++) {
+      intermediate.infections_ts(a, g) =
+          intermediate.hiv_negative_pop(a, g) * intermediate.incidence_rate_sex(g) *
+          pars.incidence_relative_risk_age(a - pars.adult_incidence_first_age_group, g, time_step) *
+          intermediate.hiv_neg_aggregate(g) /
+          intermediate.Xhivn_incagerr(g);
+    }
+  }
+}
+
+template<typename real_type>
+void run_disease_progression_and_mortality(int time_step,
+                                           const Parameters<real_type> &pars,
+                                           const State<real_type> &state_curr,
+                                           State<real_type> &state_next,
+                                           IntermediateData<real_type> &intermediate,
+                                           int hiv_step) {
+  for (int g = 0; g < pars.num_genders; g++) {
+    for (int ha = 0; ha < pars.age_groups_hiv; ha++) {
+      for (int hm = 0; hm < pars.disease_stages; hm++) {
+        // TODO: Mortality scaling not yet implemented
+        if (pars.scale_cd4_mortality &
+            (time_step >= pars.time_art_start) &
+            (hm >= intermediate.everARTelig_idx) &
+            (state_next.hiv_strat_adult(hm, ha, g) > 0.0)) {
+          for (int hu = 0; hu < pars.treatment_stages; hu++) {
+            intermediate.artpop_hahm += state_next.art_strat_adult(hu, hm, ha, g);
+          }
+          intermediate.cd4mx_scale = state_next.hiv_strat_adult(hm, ha, g) /
+                                     (state_next.hiv_strat_adult(hm, ha, g) + intermediate.artpop_hahm);
+        }
+
+        intermediate.deaths =
+            intermediate.cd4mx_scale * pars.cd4_mortality(hm, ha, g) * state_next.hiv_strat_adult(hm, ha, g);
+        intermediate.hiv_deaths_age_sex(ha, g) += pars.dt * intermediate.deaths;
+        state_next.aids_deaths_no_art(hm, ha, g) += pars.dt * intermediate.deaths;
+        intermediate.grad(hm, ha, g) = -intermediate.deaths;
+      }
+
+      for (int hm = 1; hm < pars.disease_stages; hm++) {
```

Comment:
why does this exclude hm = 0 - there's a `hm - 1` below which provides the immediate reason, but why no boundary condition